Plots/Boxplots/NCL_box_3.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+from geocat.viz import util as gvutil
+
+###############################################################################
+# Generate fake data
+
+np.random.seed(200)
+data = np.random.lognormal(size=(40, 3), mean=1, sigma=.7)
+for a in range(len(data)):
+    data[a] = [x-4 for x in data[a]]
+
+###############################################################################
+# Helper function to set edge color of boxes
+
+
+def setBoxColor(boxplot, number, edge_color):
+
+    # Set edge color of the outside of the boxes and the median lines
+    for element in ['boxes', 'medians']:
+        plt.setp(boxplot[element][number], color=edge_color)
+
+
+###############################################################################
+# Helper function to remove axis "spines" on the top and right sides
+
+
+def removeSpines(ax):
+
+    ax.spines['right'].set_visible(False)
+    ax.spines['top'].set_visible(False)
+
+
+###############################################################################
+# Plot:
+
+# Create figure and axis
+fig, ax = plt.subplots(figsize=(6, 6))
+
+# Plot each boxplot, set tick labels, and determine box widths
+boxplots = ax.boxplot(data, labels=['Control', '-2Xna', '2Xna'],
+                      widths=[0.4, 0.4, 0.4], showfliers=False)
+
+# Set whisker style to dashed
+plt.setp(boxplots['whiskers'], linestyle='--')
+
+# Set boxplot edge colors
+setBoxColor(boxplots, 0, 'blue')
+setBoxColor(boxplots, 1, 'red')
+setBoxColor(boxplots, 2, 'limegreen')
+
+# Use geocat.viz.util convenience function to set axes tick values
+gvutil.set_axes_limits_and_ticks(ax, ylim=(-6.0, 9.0),
+                                 yticks=[-3.0, 0.0, 3.0, 6.0])
+
+# Use geocat.viz.util convenience function to add minor and major tick lines
+gvutil.add_major_minor_ticks(ax, y_minor_per_major=3, x_minor_per_major=1,
+                             labelsize=14)
+
+# Use geocat.viz.util convenience function to add title to the plot axis.
+gvutil.set_titles_and_labels(ax, maintitle='Box Plot with Polymarkers')
+
+# Make both major and minor ticks point inwards towards the plot
+ax.tick_params(direction="in", which='both')
+
+# Get rid of right and top axis spines
+removeSpines(ax)
+
+# Set ticks only at left and bottom sides of plot
+ax.yaxis.set_ticks_position('left')
+ax.xaxis.set_ticks_position('bottom')
+
+# Add another partially transparent axis on top of the first one
+ax2 = ax.inset_axes([0, 0, 1, 1])
+ax2.patch.set_alpha(0.2)
+
+# Set limits of second axis so markers will line up with boxes on boxplot
+ax2.set_xlim(0, 6)
+ax2.set_ylim(-6, 9)
+
+# Turn both major and minor ticks in overlayed axis off
+ax2.tick_params(which='both', top=False, bottom=False, left=False, right=False,
+                labelleft=False, labelbottom=False)
+
+# Get rid of right and top axis spines
+removeSpines(ax2)
+
+# Plot red x markers
+ax2.scatter(1, 7.7, marker='x', color='red', linewidth=.5, s=100)
```

Comment:
Are these Xs supposed to represent the outliers or something else that is statistically significant? If these are representative of outliers, `plt.boxplot` should take care of them for you. You shouldn't have to manually plot them.

Reply:
It looks like in the NCl script the coordinates of each polymarker are manually set, so I did the same. I'm not sure how common it is to want to manually plot markers on a boxplot though, I feel like it would be more useful to plot a statistically significant point on there instead.

Reply:
After looking at the NCL script, it does seem that those points are arbitrary. I wouldn't be too concerned with it then